### ISSUE TITLE
Use literal union instead of | string

### ISSUE THREAD

**Issue by JCQuintas** [frenic/csstype#73]
This would allow typescript to autocomplete parameters.

Right now the following type doesn't autocomplete because the `string` type takes over the entire type

```typescript
type TransitionTimingFunction =
  | 'ease'
  | 'ease-in'
  | 'ease-out'
  | 'ease-in-out'
  | 'linear'
  | 'step-start'
  | 'step-end'

// No autocomplete 
const ttf: TransitionTimingFunction  | string = '' 

type LiteralUnion<T extends U, U> = T | (U & { })

// Autocompletes and allow custom strings!
const attf: LiteralUnion<TransitionTimingFunction, string> = '' 
```

[Playground](https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=14&pc=107#code/C4TwDgpgBAKgTgQwHYGcCWw0HskzQWzSQHMAxAVyQGNMcoBeAWACgooAfKAcggRQi4s2nHnwgBaIoNYduvfuKzlg04XLGSki5atlcANkV5xdIlMAhhx5hHBVC95y+IhIAJtJZUc5qMGAAZgBcsIioGNi4BERklDSRauZwMQzcXFAA9BlQAHJYUAjKWN74YPoQFlAsLKCQUAAyGBCI+gCqSJEAPDBQEAAeFu4oUK0ANCMAfKk9nAAUrVAAZFAA3lAAvgCU1czeqMAF-sENTS3tXfDI6LRRhCQU1DfjSTFT9GmZ2QCCRSVlFRBhsg3AV9PosAB3KBUcjmLD4KAvEgoACEQA)

**Comment by frenic** [comment]
Thank you! I've added a PR where I try this out and it just worked. See #74

**Comment by frenic** [comment]
Fixed in #74

**Comment by frenic** [comment]
Just noticed a problem with type guards using this hack:

```ts
function a(arg: (string & {}) | 'a' | 'b'): 'a' {
  if (arg === 'a') {
    return arg; // Type '(string & {}) | "a"' is not assignable to type '"a"'
  }
}
```
